### PR TITLE
107 timer

### LINKED ISSUE #107: [IMPLEMENT] Timer
# Feature Name

Recipe Timer

# User Story

*As I'm cooking, I would like to have a simple stopwatch-style timer to see how long I have been cooking, and approximately how far along the recipe I am.*

# Additional Context

You will have full discretion over how and where the timer is, and how it works. Just as an example (you may choose to do something else), you can have a simple stopwatch timer that displays time elapsed over recipe's expected time (i.e. 2min/40 min cook time), and simply nest this at the top of the expanded recipe page.

### PR DESCRIPTION
**Pull request by AntoniaY31**
## New Changes
- Achieve fundamental functionalities of timer: set timer, pause, reset
- Allow maximize and minimize the timer
- Allow drag and move the timer on the pc
- Complete the styling of timer

## Linked Issue
Closes #107 

## Screenshots
Please include screenshots of any visual changes to the project (new HTML/CSS)
In slack#cse110-project

## Checklist
 If it doesn't apply, just mark as completed
- [x] I have requested 2+ code reviewers
- [x] I have reviewed and commented my own code.
- [x] My code follows the style guidelines of this project.
- [x] I have added new tests as necessary.
- [x] All new and existing tests pass successfully.

### CODE REVIEW

File: source/demo-code/recipe-info-rough/recipe-page.css [hunk 1]
```diff
@@ -393,3 +394,41 @@ li.item {
...
+  background-color: #fff0d4;
+  border-radius: 10px;
+}
+
+/* stylelint-disable-next-line selector-class-pattern */
+div.timeInput {
+  display: inline-flex;
+}
+
+#timer {
```
Comment (farisashai) [inline source/demo-code/recipe-info-rough/recipe-page.css]: use class names for css pls

File: source/demo-code/recipe-info-rough/recipe-page.css [hunk 2]
```diff
@@ -393,3 +394,41 @@ li.item {
...
+  bottom: 0;
+  display: block;
+  float: left;
+  height: 140px;
+  left: 0;
+  margin: 1rem;
+  overflow: auto;
+  padding: 1rem;
+  position: fixed;
+  width: auto;
```
Comment (farisashai) [inline source/demo-code/recipe-info-rough/recipe-page.css:269]: maybe nice to also add`cursor: grab;` on hover

File: source/demo-code/recipe-info-rough/recipePage.html [hunk 3]
```diff
@@ -128,5 +129,19 @@
...
+            <input type="number" name="minutes" id="minutes" placeholder="Minutes" min="0" max="59" size = "5" maxlength="2" value="0">
+            <input type="number" name="seconds" id="seconds" placeholder="Seconds" min="0" max="59" size = "5" maxlength="2" value="0">
+            <button href="javascript:void(0);" class="setTime" id="setTime" onclick="setTimer()">Set Timer</button>
+        </div>
+        
+        <p id="timer">Begin Timer</p>
+        <button id="pause" onclick="pauseTimer()">Pause</button>
+        <button id="reset" onclick="resetTimer()">Reset</button>
+        <button id="shown" onclick="displayTimer()">Minimize</button>
+        <audio class="hidden" src=""></audio>
```
Comment (farisashai) [inline source/demo-code/recipe-info-rough/recipePage.html]:
```
var audioElement = new Audio('sound.mp3');
audioElement.play()
```
Might be easier to directly play audio from JavaScript when the timer finishes and we don't need to create an element.

**Comment by treeebooor** [comment]
The timer default values are all 0 right now and it doesn't show what units they are. Can you change it so it shows hours, minutes, seconds?

File: source/demo-code/recipe-info-rough/recipe-page.css (hunk 2 above)
Comment (AntoniaY31) [reply source/demo-code/recipe-info-rough/recipe-page.css:269]: Added

File: source/demo-code/recipe-info-rough/recipe-page.css (hunk 1 above)
Comment (AntoniaY31) [reply source/demo-code/recipe-info-rough/recipe-page.css]: FIxed

File: source/demo-code/recipe-info-rough/recipePage.html (hunk 3 above)
Comment (AntoniaY31) [reply source/demo-code/recipe-info-rough/recipePage.html]: Changed

**Comment by AntoniaY31** [comment]
> 

I put "hr", "min", "sec" as the placeholders for the input box!

**Review by farisashai — APPROVED** [review]

**Pull request merged**